File: Programmers/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int judge(int t)
{
    if(t >= 6)
        return 1;
    else if(t == 5)
        return 2;
    else if(t == 4)
        return 3;
    else if(t == 3)
        return 4;
    else if(t == 2)
        return 5;
    else return 6;
}

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    int mx = 0, mn = 0;
    for(int i = 0 ; i < lottos.size(); i++)
    {
        if(lottos[i]==0) mx++;
        for(int j = 0 ; j < win_nums.size(); j++)
        {
            if(lottos[i] == win_nums[j]) {
                mn++;
            }
        }
    }
    mx = mx + mn;
    mx = judge(mx);
    mn = judge(mn);
    answer.push_back(mx);
    answer.push_back(mn);
    return answer;
}

File: Programmers/로또의 최고 순위와 최저 순위.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
    	int mx = 0, mn = 0;
        for(int i = 0; i < lottos.length; i++) {
        	for(int j = 0; j < lottos.length; j++) 
        		if(lottos[i] == win_nums[j])
        			mn++;
        		else if(lottos[i] == 0) {
        			mx++;
                    break;
                    }
        			
        }
        mx = mx+mn;
        answer[0] = judge(mx);
        answer[1] = judge(mn);
        return answer;
    }
    int judge(int n) {
    	if(n == 6)
    		return 1;
    	else if(n == 5)
    		return 2;
    	else if(n == 4)
    		return 3;
    	else if(n == 3)
    		return 4;
    	else if(n == 2)
    		return 5;
    	return 6;
    }
}
